addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/DeviceHandler.java

```diff
@@ -172,38 +176,62 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                 break;
             case CHANNEL_SWITCH:
                 if (command instanceof OnOffType) {
-                    fritzBox.setSwitch(ain, command.equals(OnOffType.ON) ? true : false);
+                    state.getSwitch().setState(OnOffType.ON.equals(command) ? SwitchModel.ON : SwitchModel.OFF);
+                    fritzBox.setSwitch(ain, OnOffType.ON.equals(command) ? true : false);
+                } else {
+                    logger.warn("Received unknown command {} for channel {}", command.toString(), CHANNEL_SWITCH);
```

Comment:
You don't have to call toString it will be called automatically, but only when the logging-level actually applies.